### PR TITLE
fix: pass in update msg

### PR DESCRIPTION
**Pull request by Shivansh-yadav13**
Need for extending watcherEx 
need public constructor for extending
please review if this is good change?

### CODE REVIEW

**Comment by hsluoyz** [comment]
@nodece @Zxilly plz review

File: src/watcher.ts [hunk 1]
```diff
@@ -84,12 +84,12 @@ export class RedisWatcher implements Watcher {
...
 
-  private constructor() {}
+  public constructor() {}
 
-  public async update(): Promise<boolean> {
+  public async update(msg?: string): Promise<boolean> {
     const client = await this.pubConnection.getRedisClient();
     // @ts-ignore - publish is exists.
-    await client.publish(this.channel, 'casbin rules updated');
+    await client.publish(this.channel, msg ? msg : 'casbin rules updated');
```
Comment (Zxilly) [inline src/watcher.ts:92]: Is this necessary? Maybe you can override it.

Comment (Shivansh-yadav13) [reply src/watcher.ts:92]: @Zxilly I was wondering that I would have to re-write for override like the same Redis connections and all that, so I was thinking to update this.

Comment (Shivansh-yadav13) [reply src/watcher.ts:92]: @nodece @Zxilly , what about we add watcherEx class inside this project only and then it depends on the user which one they want to use?

Comment (nodece) [reply src/watcher.ts:92]: @Shivansh-yadav13 and I talked about this offline. This PR is unnecessary, so close this PR.

**Pull request closed without merging**